e2e-tests/upgrade-proxysql/run

```diff
@@ -0,0 +1,253 @@
+                && "${TARGET_IMAGE_PROXY}" == $(kubectl_bin get pxc "${CLUSTER}" -o jsonpath='{.spec.proxysql.image}') \
+                && "${TARGET_IMAGE_HAPROXY}" == $(kubectl_bin get pxc "${CLUSTER}" -o jsonpath='{.spec.haproxy.image}') \
+                && "${TARGET_IMAGE_BACKUP}" == $(kubectl_bin get pxc "${CLUSTER}" -o jsonpath='{.spec.backup.image}') \
+                && "${TARGET_IMAGE_PMM}" == $(kubectl_bin get pxc "${CLUSTER}" -o jsonpath='{.spec.pmm.image}') \
+                && "${TARGET_IMAGE_PXC}" == $(kubectl_bin get pxc "${CLUSTER}" -o jsonpath='{.spec.pxc.image}') ]]; then
+                : Cluster images have been updated correctly
+            else
+                echo 'Cluster images have not been updated'
+                exit 1
+            fi
+            ;;
+    esac
+}
+
+function apply_rbac_gh() {
+    local operator_namespace="${OPERATOR_NS:-'pxc-operator'}"
+    local rbac="${1:-'rbac'}"
+
+    curl -s "https://raw.githubusercontent.com/percona/percona-xtradb-cluster-operator/${GIT_TAG}/deploy/${rbac}.yaml" > "${tmp_dir}/rbac_${GIT_TAG}.yaml"
+    $sed -i -e "s^namespace: .*^namespace: ${operator_namespace}^" "${tmp_dir}/rbac_${GIT_TAG}.yaml"
+    kubectl_bin apply -f "${tmp_dir}/rbac_${GIT_TAG}.yaml"
+}
+
+function deploy_operator_gh() {
+    desc 'start operator'
+    kubectl_bin apply -f "https://raw.githubusercontent.com/percona/percona-xtradb-cluster-operator/${GIT_TAG}/deploy/crd.yaml" > "${tmp_dir}/crd_${GIT_TAG}.yaml"
+
+    local rbac_yaml="rbac"
+    local operator_yaml="operator.yaml"
+    if [ -n "${OPERATOR_NS}" ]; then
+        rbac_yaml="cw-rbac"
+        operator_yaml="cw-operator.yaml"
+    fi
+    apply_rbac_gh "${rbac_yaml}" "${GIT_TAG}"
+    curl -s "https://raw.githubusercontent.com/percona/percona-xtradb-cluster-operator/${GIT_TAG}/deploy/${operator_yaml}" > "${tmp_dir}/${operator_yaml}_${GIT_TAG}.yaml"
+    $sed -i -e "s^image: .*^image: ${IMAGE}^" "${tmp_dir}/${operator_yaml}_${GIT_TAG}.yaml"
+    kubectl_bin apply -f "${tmp_dir}/${operator_yaml}_${GIT_TAG}.yaml" ${OPERATOR_NS:+-n $OPERATOR_NS}
+
+    sleep 2
+    wait_pod "$(get_operator_pod)"
+}
+
+function create_infra_gh() {
+    local ns="$1"
+
+    if [ -n "${OPERATOR_NS}" ]; then
+        create_namespace "${OPERATOR_NS}"
+        deploy_operator_gh "${GIT_TAG}"
+        create_namespace "${ns}"
+    else
+        create_namespace "${ns}"
+        deploy_operator_gh "${GIT_TAG}"
+    fi
+}
+
+function prepare_cr_yaml() {
+    local cr_yaml="$1"
+    local proxy="$2"
+
+    curl -s "https://raw.githubusercontent.com/percona/percona-xtradb-cluster-operator/${GIT_TAG}/deploy/cr.yaml" > "${cr_yaml}"
+    yq w -i "${cr_yaml}" "metadata.name" "${CLUSTER}"
+    yq w -i "${cr_yaml}" "spec.upgradeOptions.apply" "disabled"
+    yq w -i "${cr_yaml}" "spec.pxc.size" "${CLUSTER_SIZE}"
+    yq w -i "${cr_yaml}" "spec.proxysql.size" "${CLUSTER_SIZE}"
```

Comment:
what do you think about passing curls output through yq pipes, not changing the file constantly?

Reply:
makes sense, updated! thanks!